libflux/src/semantic/infer.rs

```diff
@@ -11,13 +12,19 @@ use std::ops;
 // and will be unified at some point.
 //
 #[derive(Debug, PartialEq)]
-enum Constraint {
+pub enum Constraint {
     Kind(MonoType, Kind),
     Equal(MonoType, MonoType),
 }
 
 #[derive(Debug, PartialEq)]
-struct Constraints(Vec<Constraint>);
+pub struct Constraints(Vec<Constraint>);
+
+impl Constraints {
+    pub fn from(constraints: Vec<Constraint>) -> Constraints {
```

Comment:
Maybe? This method exists so that I can keep the inner field private. I'm still trying to grok fine organization in Rust though. It might make more sense to define Constraints with the methods that produce them - namely in the `nodes.rs` file.

Reply:
Yep, I see. You can follow the `From` trait pattern here, so that you get `into` for free. I think that's more Rust idiomatic: https://doc.rust-lang.org/rust-by-example/conversion/from_into.html